portfolio/src/main/java/com/google/sps/servlets/DeleteServlet.java

```diff
@@ -0,0 +1,43 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.*;
+import com.google.gson.Gson;
+import com.google.sps.data.Post;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Deletes posts from Datastore */
+@WebServlet("/deletedata")
+public class DeleteServlet extends HttpServlet {
+
+  /**
+   * Extracts text of comment to be deleted from the request URL,
+   * retrieves all entries of kind 'post' with matching comment,
+   * deletes all retrieved entries from Datastore.
+   * Reloads index.html and scrolls to the forum section.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = request.getParameter("comment");
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query =
+      new Query("post").setFilter(
+        new Query.FilterPredicate("comment", Query.FilterOperator.EQUAL, comment));
```

Comment:
Nit: Adjust indentation in your wrapped lines.

>When line-wrapping, each line after the first (each continuation line) is indented at least +4 from the original line.
>
> When there are multiple continuation lines, indentation may be varied beyond +4 as desired. In general, two continuation lines use the same indentation level if and only if they begin with syntactically parallel elements.

([Documentation](https://google.github.io/styleguide/javaguide.html#s4.5.2-line-wrapping-indent))